docs/compilers/CSharp/Runtime Async Design.md

```diff
@@ -301,23 +314,44 @@ class C
 }
 ```
 
+Translated C#:
+
+```cs
+System.Runtime.CompilerServices.RuntimeHelpers.Await(C.M<Task>());
```

Comment:
@jakobbotsch @VSadov is it going to complicate the JIT's pattern matching significantly if we use this approach for generic methods like this? In the old specialized call version, we would have needed go to through `UnsafeAwaitAwaiterFromRuntimeAsync`, but since this now just appears to be `Task`-returning method, I'm hoping we can keep IL size down and just call `Await`.

Reply:
I think you should be fine to use `Await` in these situations. It is up to the runtime/JIT to recognize when it will be unable to optimize.

Reply:
Yes. The details of how some methods get special sibling/variant methods (and some do not) is now irrelevant from the public API point. It makes sense to allow to use `Await` with anything that returns `Task`/`ValueTask`. 

JIT will sort it out what can be optimized.